adapter.go

```diff
@@ -335,11 +394,13 @@ const (
 )
 
 func NewAdapter(reader KeyReaderAt, opts ...AdapterOption) (*Adapter, error) {
```

Comment:
For the sake of clarity, I think we should add a NewStreamingAdapter function that takes a KeyStreamerAt as an argument to clarify what we are expecting